src/helper/route/index.ts

```diff
@@ -37,23 +37,32 @@ export const matchedRoutes = (c: Context): RouterRoute[] =>
  * Get the route path registered within the handler
  *
  * @param {Context} c - The context object
+ * @param {number} index - The index of the root from which to retrieve the path, similar to Array.prototype.at(), where a negative number is the index counted from the end of the matching root. Defaults to the current root index.
  * @returns The route path registered within the handler
  *
  * @example
  * ```ts
  * import { routePath } from 'hono/route'
  *
+ * app.use('*', (c, next) => {
+ *   console.log(routePath(c)) // '*'
+ *   console.log(routePath(c, -1)) // '/posts/:id'
+ *   return next()
+ * })
+ *
  * app.get('/posts/:id', (c) => {
  *   return c.text(routePath(c)) // '/posts/:id'
  * })
  * ```
  */
-export const routePath = (c: Context): string => matchedRoutes(c)[c.req.routeIndex].path
+export const routePath = (c: Context, index?: number): string =>
+  matchedRoutes(c).at(index ?? c.req.routeIndex)?.path ?? ''
```

Comment:
It isn't easy to define the behavior if it cannot find the route. I think both `''` and `undefined` are okay. In this case, `''` is better because this way does not break that `routePath` returns value is `string` (not `string | undefined`). So this is good.